examples/albert/run_trainer.py

```diff
@@ -0,0 +1,317 @@
+        if not self.are_finite_params():
+            self.load_from_state(self.previous_state)
+            return control
+
+        self.previous_state = self.get_current_state()
+
+        if state.log_history and self.collaborative_optimizer.local_step != self.last_reported_collaboration_step:
+            self.last_reported_collaboration_step = self.collaborative_optimizer.local_step
+            statistics = [self.collaborative_optimizer.local_step,
+                          self.collaborative_optimizer.performance_ema.samples_per_second,
+                          self.collaborative_optimizer.local_samples_accumulated,
+                          state.log_history[-1]['loss']]
+
+            self.dht.store(self.collaborative_optimizer.prefix + "_metrics", subkey=self.trainer_uuid,
+                           value=statistics, expiration_time=hivemind.get_dht_time() + self.statistics_expiration,
+                           return_future=True)
+            self.last_reported_collaboration_step = self.collaborative_optimizer.local_step
+        return control
+
+    @torch.no_grad()
+    def get_current_state(self) -> Dict[str, Any]:
+        return {
+            'model': self.model.state_dict(),
+            'opt': self.collaborative_optimizer.opt.state_dict()
+        }
+
+    @torch.no_grad()
+    def load_from_state(self, state):
+        self.model.load_state_dict(state['model'])
+        self.collaborative_optimizer.opt.load_state_dict(state['opt'])
+
+    @torch.no_grad()
+    def are_finite_params(self):
+        for param in self.model.parameters():
+            if not torch.all(torch.isfinite(param)):
+                return False
+        return True
+
+
+class NoOpScheduler(LRSchedulerBase):
+    """ Dummy scheduler for transformers.Trainer. The real scheduler is defined in CollaborativeOptimizer.scheduler """
+
+    def get_lr(self):
+        return [group['lr'] for group in self.optimizer.param_groups]
+
+    def print_lr(self, *args, **kwargs):
+        if self.optimizer.scheduler:
+            return self.optimizer.scheduler.print_lr(*args, **kwargs)
+
+    def step(self):
+        logger.debug("Called NoOpScheduler.step")
+        self._last_lr = self.get_lr()
+
+    def state_dict(self):
+        return {}
+
+    def load_state_dict(self, *args, **kwargs):
+        logger.debug("Called NoOpScheduler.load_state_dict")
+
+
+def main():
+    parser = HfArgumentParser((AlbertTrainingArguments, DatasetArguments, CollaborationArguments))
+    training_args, dataset_args, collaboration_args = parser.parse_args_into_dataclasses()
+
+    collaboration_args.initial_peers = list(map(str.strip, collaboration_args.initial_peers.split(',')))
+ 
+    from logging.handlers import SysLogHandler
+    print(collaboration_args.initial_peers[0].split(":")[0])
+    syslog_handler = SysLogHandler(address=(collaboration_args.initial_peers[0].split(":")[0], 514))
+    logger.addHandler(syslog_handler)
```

Comment:
Not sure if I understand the meaning of 514 and why we need to install this handler at all

Reply:
it was a temporary solution, we'll build a more permanent one in future PRs